### PR TITLE
Fix uncaught IllegalArgumentException when reading broken WAL files

### PR DESCRIPTION
**Pull request by jt2594838**
As the title indicates.

### CODE REVIEW

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALMetaData.java [hunk 1]
```diff
@@ -131,38 +131,45 @@ public long getFirstSearchIndex() {
...
+        for (int size : metaData.buffersSize) {
+          channel.position(offset);
+          ByteBuffer buffer = ByteBuffer.allocate(Long.BYTES);
+          channel.read(buffer);
+          buffer.clear();
+          metaData.memTablesId.add(buffer.getLong());
+          offset += size;
+        }
       }
+    } catch (IllegalArgumentException e) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALMetaData.java:170]: maybe we can catch Exception?

Comment (jt2594838) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALMetaData.java:170]: Done

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**